src/shogun/lib/observers/ObservedValueTemplated.h

```diff
@@ -21,6 +21,23 @@ namespace shogun
 	{
 
 	public:
+		/**
+		 * Constructor
+		 * @param step step
+		 * @param name the observed value's name
+		 * @param value the observed value
+		 */
+		ObservedValueTemplated(
+		    const int64_t step, const std::string_view name, const T value)
+		    : ObservedValue(step, name), m_observed_value(value)
+		{
+			this->watch_param(
+			    name, &m_observed_value,
+			    AnyParameterProperties(
+			        "Unamed observed value", ParameterProperties::READONLY));
```

Comment:
un-named? (also there is a type)

Reply:
I'm not sure about this constructor. I may change it later on.